### DISCUSSION TITLE
Welcome to gpiozero Discussions!

### DISCUSSION THREAD

**Discussion by bennuttall** [gpiozero/gpiozero#906] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:

  * Ask questions
  * Share ideas
  * Show us what you've built with gpiozero
  * Engage with other community members and offer help
  * Welcome others

**Comment by bcroston** [comment]
Ben Croston - maintainer of RPi.GPIO which is the default pin factory of gpiozero.

**Reply by JonnyAlpha** [reply]
According to the docs it's BCM https://gpiozero.readthedocs.io/en/stable/recipes.html#pin-numbering

**Reply by waveform80** [reply]
Just to clear up any confusion:

* Ben's correct: the default *pin factory* is RPi.GPIO; BCM is a pin numbering system and
* Jonny's correct that BCM is the default pin *numbering system* in gpiozero

In fact, fundamentally BCM is the *only* numbering system we support; we do provide a translation layer for other systems but ultimately they all become BCM under the covers, whereas pin factories are more complex and are global state (to some degree - changing pin factory at runtime is non-trivial and not something we really support).

**Comment by lurch** [comment]
Andrew Scheller - I did a lot of work on GpioZero itself along with @bennuttall and @waveform80 

> Ask questions you’re wondering about.

When will the next stable [release](https://github.com/gpiozero/gpiozero/releases) of gpiozero get made? :wink:

**Reply by waveform80** [reply]
"Shortly" ;) (honest!)

**Comment by JonnyAlpha** [comment]
Hi, 
I am using a Raspberry Pi to control 2 x DC motors that power a Nerf launcher. The motors need to be switched on / off simultaneously  and only run in one direction. The circuit uses GPIO Pin 27 connected to the gate of a MOSFET to turn the motors (connected in parallel) on and off. I am using gpiozero LED in my python code which to drive the pin high and low, this seems to work fine. If I also wanted to control the speed of the motors using PWM could I use PWMLED? Using LED works but is there another command in gpiozero that I should be using, if not should there be a general command to turn a pin high / low?

**Reply by lurch** [reply]
> The circuit uses GPIO Pin 27 connected to the gate of a MOSFET to turn the motors (connected in parallel) on and off.

There's also https://gpiozero.readthedocs.io/en/stable/recipes.html#motors (which needs an H-bridge rather than just a MOSFET) if you wanted to spin your motor(s) in both directions.

> If I also wanted to control the speed of the motors using PWM could I use PWMLED?

Yes that should work, and is the same strategy that the [Motor class](https://gpiozero.readthedocs.io/en/stable/api_output.html#gpiozero.Motor) uses for speed control.

> Using LED works but is there another command in gpiozero that I should be using

`LED` is just a wrapper around [DigitalOutputDevice](https://gpiozero.readthedocs.io/en/stable/api_output.html#gpiozero.DigitalOutputDevice) and `PWMLED` is just a wrapper around [PWMOutputDevice](https://gpiozero.readthedocs.io/en/stable/api_output.html#gpiozero.PWMOutputDevice), so you can use those classes directly if you'd prefer :slightly_smiling_face:

**Reply by JonnyAlpha** [reply]
Great thanks

**Comment by JonnyAlpha** [comment]
Thanks for all the help on this. Not tried PWM yet, in fact I am now considering a different launcher :-)
